app/services/XmlBuilderService.scala

```diff
@@ -132,15 +132,36 @@ class XmlBuilderService {
                 buildAndEncodeElem(event.alreadyInNcts,"AlrInNCTCTL29")
               }
             </CTLCTL> {
-               buildIncident(event.eventDetails)(arrivalNotificationRequest)
+               buildIncident(event.eventDetails, event.seals)(arrivalNotificationRequest)
             }
           </ENROUEVETEV>
       }
   }
 
-  private def buildIncident(event: EventDetails)(implicit arrivalNotificationRequest: ArrivalNotificationRequest): NodeSeq = event match {
-    case incident: Incident =>
-      <INCINC> {
+  private def buildSeals(seals: Option[Seq[Seal]]): NodeSeq =
```

Comment:
I think we need to invert the control here, it should be the concern of the `buildSeals` function to construct a valid `NodeSeq` given a `Seq[Seal]`. In its current form, it is also introspecting on its parameter (`seals: Option[Seq[Seal]] `) and checking if it needs to default, which feels like a concern of the orchestrator which is the `buildIncident` method.

If we do this though, we would need to move the defaulting of the `NodeSeq.Empty` to the place where this method is called